### PR TITLE
Deploy as an app CR in control planes

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#7896

This completes the flattening code changes. Needs https://github.com/giantswarm/app-operator/pull/255 to be merged first. 

Once this is merged we'll create a 0.13.0 release. We want to use major version 1 to distinguish chart-operator CRs using Helm 3.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -86,16 +59,20 @@ workflows:
...
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "chart-operator"
           requires:
-            - build
+            - push-chart-operator-to-quay
+          filters:
+            # Needed to trigger job also on git tag.
+            tags:
+              only: /^v.*/
```
Comment (rossf7) [inline .circleci/config.yml:72]: We now push to the control plane and default catalogs.

File: .circleci/config.yml [hunk 2]
```diff
@@ -86,16 +59,20 @@ workflows:
...
 
       - architect/push-to-app-catalog:
-          name: push-chart-operator-to-app-catalog
+          name: push-chart-operator-to-default-app-catalog
```
Comment (rossf7) [inline .circleci/config.yml:75]: Just fixing the name.

File: helm/chart-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ spec:
     matchLabels:
       {{- include "chart-operator.selectorLabels" . | nindent 6 }}
   strategy:
-    type: RollingUpdate
+    type: Recreate
```
Comment (rossf7) [inline helm/chart-operator/templates/deployment.yaml:15]: As per the flattening docs.

File: .circleci/config.yml [hunk 4]
```diff
@@ -106,6 +83,45 @@ workflows:
             tags:
               only: /^v.*/
 
+      - architect/push-to-app-collection:
+          name: push-chart-operator-to-aws-app-collection
+          app_name: "chart-operator"
+          app_collection_repo: "aws-app-collection"
+          unique: "true"
```
Comment (rossf7) [inline .circleci/config.yml:90]: We use the unique flag.

File: .circleci/config.yml [hunk 5]
```diff
@@ -106,6 +78,45 @@ workflows:
             tags:
               only: /^v.*/
 
+      - architect/push-to-app-collection:
```
Comment (kopiczko) [inline .circleci/config.yml:81]: You can add ‘context: architect’ and remove variables from project settings in the web console.

File: .circleci/config.yml [hunk 6]
```diff
@@ -106,6 +78,45 @@ workflows:
...
+          name: push-chart-operator-to-aws-app-collection
+          app_name: "chart-operator"
+          app_collection_repo: "aws-app-collection"
+          unique: "true"
+          requires:
+            - push-chart-operator-to-control-plane-app-catalog
```
Comment (kopiczko) [inline .circleci/config.yml:87]: ~~Recently we decided to wait for aliyun too.~~

File: .circleci/config.yml [hunk 7]
```diff
@@ -86,16 +54,20 @@ workflows:
...
-          username_envar: "ALIYUN_USERNAME"
-          password_envar: "ALIYUN_PASSWORD"
+      - architect/push-to-app-catalog:
+          name: push-chart-operator-to-control-plane-app-catalog
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "chart-operator"
           requires:
-            - build
+            - push-chart-operator-to-quay
```
Comment (kopiczko) [inline .circleci/config.yml:63]: Recently we decided to wait for aliyun too.

File: .circleci/config.yml (hunk 5 above)
Comment (rossf7) [reply .circleci/config.yml:85]: Thx. Done. I'll remove from the console once merged.

File: .circleci/config.yml (hunk 7 above)
Comment (rossf7) [reply .circleci/config.yml:66]: Ack. Done.

**Pull request merged**